core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/domain/solution/cloner/FieldAccessingSolutionCloner.java

```diff
@@ -53,320 +53,280 @@ public FieldAccessingSolutionCloner(SolutionDescriptor<Solution_> solutionDescri
 
     @Override
     public Solution_ cloneSolution(Solution_ originalSolution) {
-        return new FieldAccessingSolutionClonerRun(originalSolution)
-                .call();
+        int entityCount = solutionDescriptor.getEntityCount(originalSolution);
+        Map<Object, Object> originalToCloneMap = new IdentityHashMap<>(entityCount + 1);
+        Queue<Unprocessed> unprocessedQueue = new ArrayDeque<>(entityCount + 1);
+        Solution_ cloneSolution = clone(originalSolution, originalToCloneMap, unprocessedQueue);
+        while (!unprocessedQueue.isEmpty()) {
+            Unprocessed unprocessed = unprocessedQueue.remove();
+            Object cloneValue = process(unprocessed, originalToCloneMap, unprocessedQueue);
+            AbstractFieldCloner.setGenericFieldValue(unprocessed.bean, unprocessed.field, cloneValue);
+        }
+        validateCloneSolution(originalSolution, cloneSolution);
+        return cloneSolution;
     }
 
-    /**
-     * This method is thread-safe.
-     *
-     * @param clazz never null
-     * @param <C> type
-     * @return never null
-     */
     @SuppressWarnings("unchecked")
     private <C> Constructor<C> retrieveCachedConstructor(Class<C> clazz) {
-        return (Constructor<C>) constructorMemoization.computeIfAbsent(clazz, key -> {
-            Constructor<C> constructor;
-            try {
-                constructor = clazz.getDeclaredConstructor();
-            } catch (ReflectiveOperationException e) {
-                throw new IllegalStateException("The class (" + clazz
-                        + ") should have a no-arg constructor to create a planning clone.", e);
-            }
-            constructor.setAccessible(true);
-            return constructor;
-        });
+        synchronized (constructorMemoization) {
```

Comment:
How much faster is this single-threaded and how much slower is multi-threaded?

Reply:
I don't have metrics for either. The change needed to be done.
The entire PR is ~30 % faster single-threaded.